### PR TITLE
Add encryption methods to ethclient

### PR DESCRIPTION
**Pull request by bakaoh**
Fix: #17816

### CODE REVIEW

File: ethclient/eip1024.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+// EncryptedData is encrypted blob
+type EncryptedData struct {
+	Version        string `json:"version,omitempty"`
+	Nonce          string `json:"nonce,omitempty"`
+	EphemPublicKey string `json:"ephemPublicKey,omitempty"`
+	Ciphertext     string `json:"ciphertext,omitempty"`
+}
+
+// GetEncryptionPublicKey returns user's public Encryption key derived from privateKey Ethereum key
+func (ec *Client) GetEncryptionPublicKey(ctx context.Context, receiverAddress string) string {
```
Comment (decentralisedkev) [inline ethclient/eip1024.go]: Could receiverAddress be changed to something like ethPrivKey?

File: ethclient/eip1024.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	return base64.StdEncoding.EncodeToString(publicKey[:])
+}
+
+// Encrypt plain data
+func (ec *Client) Encrypt(ctx context.Context, receiverPublicKey string, data []byte, version string) (*EncryptedData, error) {
+	switch version {
+	case "x25519-xsalsa20-poly1305":
+		ephemeralPublic, ephemeralPrivate, _ := box.GenerateKey(rand.Reader)
+
+		publicKey0, _ := base64.StdEncoding.DecodeString(receiverPublicKey)
```
Comment (decentralisedkev) [inline ethclient/eip1024.go]: Is there a reason why you omit the errors here?

File: ethclient/eip1024.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+	default:
+		return nil, errors.New("Encryption type/version not supported")
+	}
+}
+
+// Decrypt some encrypted data.
+func (ec *Client) Decrypt(ctx context.Context, receiverPrivatekey string, encryptedData *EncryptedData) ([]byte, error) {
+	switch encryptedData.Version {
+	case "x25519-xsalsa20-poly1305":
+		privateKey0, _ := hexutil.Decode("0x" + receiverPrivatekey)
```
Comment (decentralisedkev) [inline ethclient/eip1024.go]: Is there a reason why errors are omitted here?

**Comment by decentralisedkev** [comment]
Comparing it against the js implementation, the functionality looks good to me. Was there any reason to not add the test cases used in the js implementation?

For reference: https://github.com/topealabi/eth-sig-util/blob/master/test/index.js

This should ensure cross compatibility, as other libraries will be using the js as a reference

**Comment by fjl** [comment]
Before you put any more effort into this PR: we will not add this functionality to
ethclient or to go-ethereum, for the following reasons:

- Package ethclient is for talking to web3 JSON-RPC API servers.
  The methods you have added don't need an RPC server.
- In fact no code from go-ethereum is required for this new API. Your implementation
  uses `hexutil.Decode`, which could be replaced with `hex.DecodeString`.

The code you added is essentially a wrapper around golang.org/x/crypto/nacl/box with a
string-based API. If you feel such API is useful to you and others you can just create a
new repository containing the wrapper. Ethereum-related Go code that wants to use EIP-1024
can import your package alongside ethclient and it will be just as convenient.

**Pull request closed without merging**